Add support for decompiling OOTB bin jars and attaching sources

MihaiSprinceana opened the pull request:
Signed-off-by: Mihai Sprinceana sprinceana.mihai87@gmail.com

MihaiSprinceana committed:
Add support for decompiling OOTB bin jars and attaching sources

mlytvyn reviewed (commented):
@MihaiSprinceana , thank you for your contribution. Most probably there will be few iterations of review, so some comments may become irrelevant, new may appear.

MihaiSprinceana committed:
Addressed PR comments.

Switch decompiled sources control from application-level to project-level settings

mlytvyn commented on modules/java/decompiler/src/sap/commerce/toolset/java/decompilation/DecompilerService.kt:
```diff
@@ -0,0 +1,88 @@
+ * Service to wrap IdeaDecompiler.
+ */
+@Service(Service.Level.APP)
+class DecompilerService {
```
to ensure that `IdeaDecompiler` instance will be cleared out after decompilation operation I suggest to limit purpose of the class to only "DecompilerConsentService". and move `fun decompile` to the configurator, so instance of the `IdeaDecompiler` will be created only during configuration and cleared by GC after processing.

mlytvyn commented on modules/java/core/src/sap/commerce/toolset/java/configurator/DecompiledOotbSourcesConfigurator.kt:
```diff
@@ -0,0 +1,343 @@
+                ) return@flatMap emptyList()
+
+                val moduleRoot = descriptor.path.takeIf { it.isNotEmpty() }?.let { File(it) } ?: return@flatMap emptyList()
+                val hasSources = ProjectConstants.Directory.SRC_DIR_NAMES
```
it may happen that `.jar` has testsrc, but no `src`, what it we will process all jars, but decompile only those classes which are not available in any of the src directories: `src`, `testsrc`, `groovysrc`, `groovytestsrc`, `kotlinsrc`, `kotlintestsrc`?
